qiskit/synthesis/boolean/boolean_expression.py

```diff
@@ -0,0 +1,185 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A quantum oracle constructed from a logical expression or a string in the DIMACS format."""
+
+import ast
+import itertools
+import re
+from os.path import basename, isfile
+
+from qiskit.circuit import Gate
+from .boolean_expression_visitor import (
+    BooleanExpressionEvalVisitor,
+    BooleanExpressionArgsCollectorVisitor,
+)
+
+
+class BooleanExpression(Gate):
```

Comment:
we usually don't have gates in the synthesis library, but I don't know where to put it.
this won't be publicly available?